.github/workflows/golangci.yml

```diff
@@ -0,0 +1,46 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      - uses: actions/checkout@v3
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v3.2.0
+        with:
+          # Optional: version of golangci-lint to use in form of v1.2 or v1.2.3 or `latest` to use the latest version
+          version: latest
```

Comment:
I'm wondering if we should use a version. That way if a new version causes our code to break the linter check, we could fix it in the pr which updates the version of the linter, rather than having the linter break on a random day

Reply:
Here's my thinking on this, basically I figure it's actually better if it breaks on a random day and discovers an issue that was previously hidden. If you don't mind with that thinking that's totally fine and I can put a version on it but generally speaking I look at stuff like the linter as best if it is rolling so that we always use the latest.

Reply:
Anyhow, just lmk your thoughts, don't really mind either way-- just look at this as a possible security enhancement

Reply:
I second @faddat - if it breaks, then you can fix version and make a lint update in a following PR.

Reply:
Makes sense to me!